### PR TITLE
updating dependencies and rules to latest setup

### PR DESCRIPTION
**Pull request by 5c077yP**
heavily based on #11 

### Changes

- Upgrades dependencies to latest version 
- Adds ESLint 4 as a peer dependency (just to be safe)
- Adds Prettier config to remove `airbnb` formatting rules
- Removes custom formatting rules
- Adds Prettier to the project itself
- Improves readability of rules by using `"off"`, `"warn"` and `"error"` instead of `0`, `1`, and `2`
- incorporate changes suggested on #11 comments

### CODE REVIEW

File: rules/default.js [hunk 1]
```diff
@@ -1,29 +1,34 @@
...
+    // turned off rules
+    'no-shadow': 'off',
+    'no-param-reassign': 'off',
+    'no-underscore-dangle': 'off',
+    'import/prefer-default-export': 'off',
+    'func-names': 'off',
+    'class-methods-use-this': 'off',
+
+    // warnings
+    'prefer-arrow-callback': [
```
Comment (cangoektas) [inline rules/default.js:19]: wouldn't this only add more noise because we don't use arrow functions in the tests?

Comment (cangoektas) [reply rules/default.js:19]: Sorry, I see it was a warning before already. I'd say we should turn it off 👍

**Comment by cangoektas** [comment]
Could you give a short changelog of what rules you changed? From what I can see not all suggestions from #11 were taken into account.

**Comment by sodoku** [comment]
After talking about this with @5c077yP and @cangoektas, we want to do the following:

1. Change warnings to errors
2. Check how many errors we get in app and cc
3. If there are not many errors: fix them and leave warnings as errors
    If there are a lot of errors: turn warning rules off

**Review by sodoku — APPROVED** [review]

**Comment by 5c077yP** [comment]
@cangoektas i think i took all of them, what's missing?

**Comment by 5c077yP** [comment]
will merge this in now, but i'm still open for further adjustments, just want to start integrating it now

**Pull request merged**